src/main/java/digital/slovensko/autogram/core/eforms/EFormUtils.java

```diff
@@ -154,15 +181,32 @@ public static String getFormUri(Element xdc) {
         if (xmlData == null)
             return null;
 
-        return xmlData.getAttributes().getNamedItem("Identifier").getNodeValue();
+        var identifierNode = xmlData.getAttributes().getNamedItem("Identifier");
+
+        if (identifierNode != null)
+            return identifierNode.getNodeValue();
+
+        var firstChild = xmlData.getFirstChild();
+        if (firstChild == null)
+            return null;
+
+        var xsiSchemaLocationNode = firstChild.getAttributes().getNamedItem("xsi:schemaLocation");
+        if (xsiSchemaLocationNode == null)
+            return null;
+
+        return xsiSchemaLocationNode.getNodeValue();
     }
 
     public static String getNamespaceFromEformXml(Node xml) {
         var xmlns = xml.getAttributes().getNamedItem("xmlns");
-        if (xmlns == null)
-            return null;
+        if (xmlns != null)
+            return xmlns.getNodeValue();
+
+        xmlns = xml.getAttributes().getNamedItem("xsi:schemaLocation");
```

Comment:
Aj tuto to nedratujme na namespace alias ale pouzime getNameItemNS - lebo ak sa to v xml premenuje (ten NS), tak to uz nebude fungovat.

Reply:
To NS akosi nefunguje s týmito atribútmi. Asi robím niečo zle. Skúšal som `getNamedItemNS("xsi", "schemaLocation")`. Každopádne, toto je len pre ORSR formuláre. Pre ostatné je údaj už v xmlns.

Reply:
okay, už som zistil, že som tam dával blbosť. Namiesto xsi má ísť samozrejme uri tej schémy.